homeassistant/components/monzo/__init__.py

```diff
@@ -0,0 +1,328 @@
+            _LOGGER.error(
+                "External error handling pot transfer: one or more transactions failed"
+            )
+
+    async def register_webhook(
+        call_or_event_or_dt: ServiceCall | Event | datetime | None,
+    ) -> None:
+        for acc in hass.data[DOMAIN][entry.entry_id]["accounts"]:
+            webhook_id = entry.entry_id + acc["id"]
+            if (
+                "webhooks_ids" not in entry.data
+                or webhook_id not in hass.data[DOMAIN][entry.entry_id]["webhook_ids"]
+            ):
+                hass.data[DOMAIN][entry.entry_id]["webhook_ids"].append(webhook_id)
+
+            if cloud.async_active_subscription(hass):
+                webhook_url = await async_cloudhook_generate_url(
+                    hass, entry, webhook_id
+                )
+            else:
+                webhook_url = webhook_generate_url(hass, webhook_id)
+
+            if not webhook_url.startswith("https://"):
+                _LOGGER.warning(
+                    "Webhook not registered - "
+                    "https and port 443 is required to register the webhook"
+                )
+                return
+
+            webhook_register(
+                hass,
+                DOMAIN,
+                "Monzo",
+                webhook_id,
+                async_handle_webhook,
+            )
+
+            try:
+                await externalapi.user_account.register_webhooks(webhook_url)
+                _LOGGER.info("Register Monzo webhook: %s", webhook_url)
+            except InvalidMonzoAPIResponseError:
+                _LOGGER.error("Error during webhook registration")
+            else:
+                entry.async_on_unload(
+                    hass.bus.async_listen_once(
+                        EVENT_HOMEASSISTANT_STOP, unregister_webhook
+                    )
+                )
+
+    async def unregister_webhook(
+        call_or_event_or_dt: ServiceCall | Event | datetime | None,
+    ) -> None:
+        if "webhook_ids" not in entry.data:
+            return
+        for webhook_id in hass.data[DOMAIN][entry.entry_id]["webhook_ids"]:
+            _LOGGER.debug("Unregister Monzo webhook (%s)", webhook_id)
+            async_dispatcher_send(
+                hass,
+                f"signal-{DOMAIN}-webhook-None",
+                {"type": "None", "data": {WEBHOOK_PUSH_TYPE: WEBHOOK_DEACTIVATION}},
+            )
+            webhook_unregister(hass, webhook_id)
+            try:
+                await externalapi.user_account.unregister_webhooks()
+            except InvalidMonzoAPIResponseError:
+                _LOGGER.debug(
+                    "No webhook to be dropped for %s",
+                    webhook_id,
+                )
+
+    session = config_entry_oauth2_flow.OAuth2Session(hass, entry, implementation)
+
+    externalapi = AsyncConfigEntryAuth(
+        aiohttp_client.async_get_clientsession(hass), session
+    )
+
+    coordinator = DataUpdateCoordinator(
```

Comment:
Consider extracting the coordinator into a `coordinator.py`. I will greatly improve code quality

Reply:
I'd agree if I was creating my own coordinator, but there's nothing to extract here really. Checking around, I'm only finding integrations which have their own derived coordinator with `coordinator.py` files, while integrations that just use the basic `DataUpdateCoordinator` don't. Correct me if I'm misunderstanding though!